router/internal/retrytransport/retry_transport.go

```diff
@@ -101,6 +104,20 @@ func (rt *RetryHTTPTransport) RoundTrip(req *http.Request) (*http.Response, erro
 	return resp, err
 }
 
+func (rt *RetryHTTPTransport) drainBody(resp *http.Response) {
+	if resp == nil || resp.Body == nil {
+		return
+	}
+	_, err := io.Copy(io.Discard, resp.Body)
+	if err != nil {
+		rt.Logger.Error("Failed draining when copying the body", zap.Error(err))
+	}
+	err = resp.Body.Close()
```

Comment:
We probably want to do something like defer body.Close() at the beginning of the function to make sure we always at least attempt to close the body, right?

Reply:
I didn't use defer because we don't early return on errors. But nothing against moving to defer, will change.